### ISSUE TITLE
Multiple audio streams

### ISSUE THREAD

**Issue by phi1-h** [dularion/streama#225]
Hi dularion, 
congrats for creating such a great service!
I'm just starting to use streama and already loving it ;)
I have an enhancement proposal, which is to let the user choose between multiple audio streams if available.
I'm also interested in helping the development, but not experienced, yet
Greetings from Munich,
Phil

**Comment by dularion** [comment]
Hi Phil!
Thanks for your kind words :) 
About the audio stream, that's something that I definitely want for the application too! Big Apparently there is support for multiple audioTracks for html5 videos, however, I haven't tested it yet... http://gingertech.net/2011/05/01/html5-multi-track-audio-or-video/

If it's as easy as described in the link then it should be a quick addon. It just needs a bit of angular logic to display the tracks in a menu and then disable/enable on click

**Comment by phi1-h** [comment]
Nice! I definitely want to contribute just don't know angular or grails yet. Any recommendation to start learning ? 
Thanks,
Phil

**Comment by dularion** [comment]
For angularJS, check out the official angularJS homepage. they have a nice interactive tutorial. Alternatively, a guy called egghead makes amazing tutorials for AngularJS. Note that this project still uses angular 1, not 2. 

For Grails, there is no tutorial that really goes through everything in this project, which is unfortunate. although the grails docs are pretty great. maybe they got some youtube vids too. When i got more time, I'll definitely create a learning tutorial series for grails and angular.

**Comment by gkiko** [comment]
These helped me understand how streama works
[grails user guide](http://docs.grails.org/latest/)
[angular tutorial](https://www.codecademy.com/learn/learn-angularjs)

I suggest you try to refactor code or fix bugs or implement some features listed on _Issues_ tab

**Comment by phi1-h** [comment]
You're both answering incredibly fast! Thanks I'll definitely check all of them out!

**Comment by dularion** [comment]
Hm.. so in 2013 some time development into the above solution stopped, so it was never implemented in Chrome it seems. So the way to do it would be to add audio tracks as mp3 and allow language switching that way. I guess that makes most sense, also it facilitates manually correcting audio sync issues. What do you think?

**Comment by phi1-h** [comment]
I agree. Some of my movies do already have audio / video sync issues and if there was an option, as in vlc the f and g shortcut, to delay the audio stream, that would help in these cases. Can your player extract multiple audio streams into separate mp3s ? I really want to help :)

**Comment by dularion** [comment]
Yep this is indeed related to #272. I think we should try to first see if we can sync it with uploaded mp3s and then if that works reliably we can use a third party tool to extract the audio tracks.

**Comment by phi1-h** [comment]
Sounds like a plan 👍
I'm beginning to dive into angular, and getting to know your code.

**Comment by dularion** [comment]
That's awesome! Let me know if I can help

**Comment by Albatrosicks** [comment]
Is it possible to get audio & subtitles from mkv?

**Comment by dularion** [comment]
This topic is definitely on our radar still, we are just trying to figure out how to best do this. Subtitle extraction as suggested by @Albatrosicks  is also a good idea!
